core/utils_test.py

```diff
@@ -794,3 +795,144 @@ def test_url_open(self) -> None:
+                ['-1', '-2', '-3', '-4', '-5'],
+                ['-2', '-5', '-4']),
+                ['-1', '-3'])
+        self.assertEqual(utils.compute_list_difference(
+                ['-1', '-2', '-3', '-4', '-5'],
+                ['-5', '-4', '-3', '-2', '-1']),
+                [])
+        self.assertEqual(utils.compute_list_difference(
+                ['-1', '-2', '-3', '-4', '-5'],
+                ['-6', '-7', '-8', '-9', '-10']),
+                ['-1', '-2', '-3', '-4', '-5'])
+        self.assertEqual(utils.compute_list_difference(
+                ['-1', '-2'],
+                ['-1', '-2', '-3', '-4', '-5']),
+                [])
+
+    def test_convert_png_binary_to_data_url(self) -> None:
+        filepath_png = os.path.join('core', 'tests', 'data', 'test_png_img.png')
+        file_contents_png = utils.get_file_contents(
+            filepath_png, raw_bytes=True, mode='rb')
+        self.assertEqual(utils.convert_png_binary_to_data_url(file_contents_png), 'data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAcAAAAGCAIAAACAbBMhAAAAAXNSR0IArs4c6QAAAARnQU1BAACxjwv8YQUAAAAJcEhZcwAAEnQAABJ0Ad5mH3gAAAAySURBVBhXY/iPDYBEV6xY0draCuFDAEgUKMTAANUEUYFuAkQFihIIGwigosiG/P//HwD5HmjphyAmJQAAAABJRU5ErkJggg%3D%3D')  # pylint: disable=line-too-long
+
+    def test_get_exploration_components_from_dir_with_yaml_content(self) -> None: # pylint: disable=line-too-long
+        img1_path = 'images/sample_Img.svg'
+        img2_path = 'images/sample_Img2.svg'
+        img1_file_content = b'<svg width="100" height="100"><circle cx="50" cy="50" r="40" stroke="green" stroke-width="4" fill="yellow" /></svg>'  # pylint: disable=line-too-long
+        img2_file_content = b'<svg width="400" height="110"><rect width="300" height="100" style="fill:rgb(0,0,255);stroke-width:3;stroke:rgb(0,0,0)" /></svg>'  # pylint: disable=line-too-long
+        yaml_content = 'name: John Smith\ncontact:\n    home:   1012355532\n    office:  5002586256\naddress:\n  street: |\n            123 Tornado Alley\n            Suite 16            \n    city:   East Centerville\n    state:  KS'   # pylint: disable=line-too-long
+        result = utils.get_exploration_components_from_dir(
+            'core/tests/data/dummy_assets_yaml')
+        final_result = (result[0], set(result[1]))
+        self.assertEqual(
+            final_result,
+            (
+                yaml_content,
+                set(
+                    [
+                        (
+                            img2_path,
+                            img2_file_content),
+                        (
+                            img1_path,
+                            img1_file_content)
+                            ]
+                        )
+                    )
+                )
+
+    def test_get_current_time_in_millisecs_with_current_time(self) -> None:
+        time_instance1 = utils.get_current_time_in_millisecs()
+        if isinstance(time_instance1, float):
+            pass
```

Comment:
Instead of this noop, can you use something like `time.sleep` to make clear that we're waiting here? Right now, future devs might be confused about what this is doing.

Reply:
Done